sdk/scheduler/src/main/java/com/mesosphere/sdk/state/SchemaVersionStore.java

```diff
@@ -53,6 +84,15 @@ public SchemaVersionStore(Persister persister) {
      * @throws IllegalStateException if a value is present which doesn't match the expected value
      */
     public void check(int expectedVersion) throws StateStoreException {
+        int currentVersion = getOrSetVersion(expectedVersion);
```

Comment:
Can these `check()` and `getOrSetVersion()` calls be updated to accept the enum values as well?

Reply:
I tried to do that initially but then realized we won't be able to test by passing random integer values in our unit testing. I will refactor if we are happy with unit tests passing `UNKNOWN`.